### ISSUE TITLE
[Bug]: MySQL error when executing 'occ maintenance:repair'

### ISSUE THREAD

**Issue by scebbers** [nextcloud/server#42857]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

After updating to Hub 7 (Nextcloud 28), I received the following error, which I also get when trying to manually repair the database via occ maintenance:repair:

```
% php occ maintenance:repair
 - Repair MySQL collation
     - Change row format for oc_mail_messages ...
     - Change collation for oc_mail_messages ...
     - ERROR: An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes
 - Repair mime types
 - ...
```

I've studied the issue and made sure the collation of the database was set to utf8mb4_general_ci.

### Steps to reproduce

1. Upgrade to Nextcloud 28
2. Wait for upgrade to finish and update the database

### Expected behavior

Repair of MySQL collation succeeds.

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

28

### Operating system

Other

### PHP engine version

PHP 8.2

### Web server

Nginx

### Database engine version

MySQL

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 22 to 23)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "apps_paths": [
            {
                "path": "\/usr\/local\/www\/nextcloud\/apps",
                "url": "\/apps",
                "writable": true
            }
        ],
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost",
            "192.168.178.10",
            "192.168.178.12",
            "xx.xx.nl",
            "xx.xx.nl"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "overwritehost": "xx.xx.nl",
        "overwriteprotocol": "https",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "28.0.1.1",
        "default_phone_region": "NL",
        "default_language": "nl",
        "default_locale": "nl",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "filelocking.enabled": true,
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379,
            "timeout": 0
        },
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "pipe",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "maintenance": false,
        "app_install_overwrite": [
            "printer",
            "files_external_onedrive",
            "impersonate",
            "breezedark"
        ],
        "theme": "",
        "loglevel": 2,
        "debug": false,
        "updater.release.channel": "stable",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpsecure": "ssl",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.20.0
  - admin_audit: 1.18.0
  - announcementcenter: 6.7.0
  - audioplayer: 3.4.1
  - bruteforcesettings: 2.8.0
  - calendar: 4.6.3
  - circles: 28.0.0-dev
  - cloud_federation_api: 1.11.0
  - comments: 1.18.0
  - contacts: 5.5.1
  - contactsinteraction: 1.9.0
  - dashboard: 7.8.0
  - dav: 1.29.1
  - federatedfilesharing: 1.18.0
  - federation: 1.18.0
  - files: 2.0.0
  - files_external: 1.20.0
  - files_pdfviewer: 2.9.0
  - files_reminders: 1.1.0
  - files_sharing: 1.20.0
  - files_trashbin: 1.18.0
  - files_versions: 1.21.0
  - firstrunwizard: 2.17.0
  - forms: 4.0.0
  - impersonate: 1.15.0
  - integration_onedrive: 3.1.0
  - logreader: 2.13.0
  - lookup_server_connector: 1.16.0
  - mail: 3.5.3
  - maps: 1.3.1
  - nextcloud_announcements: 1.17.0
  - notes: 4.9.2
  - notifications: 2.16.0
  - oauth2: 1.16.3
  - onlyoffice: 9.0.0
  - password_policy: 1.18.0
  - photos: 2.4.0
  - previewgenerator: 5.4.0
  - printer: 0.0.5
  - privacy: 1.12.0
  - provisioning_api: 1.18.0
  - recommendations: 2.0.0
  - related_resources: 1.3.0
  - serverinfo: 1.18.0
  - settings: 1.10.1
  - sharebymail: 1.18.0
  - spreed: 18.0.1
  - support: 1.11.0
  - survey_client: 1.16.0
  - systemtags: 1.18.0
  - tasks: 0.15.0
  - text: 3.9.1
  - theming: 2.3.0
  - twofactor_backupcodes: 1.17.0
  - twofactor_totp: 10.0.0-beta.2
  - updatenotification: 1.18.0
  - user_status: 1.8.1
  - viewer: 2.2.0
  - weather_status: 1.8.0
  - workflowengine: 2.10.0
Disabled:
  - encryption: 2.16.0 (installed 2.13.0)
  - files_archive: 1.1.3 (installed 1.1.3)
  - files_external_onedrive: 1.1.0 (installed 1.1.0)
  - files_rightclick: 0.15.1 (installed 1.6.0)
  - metadata: 0.19.0 (installed 0.19.0)
  - suspicious_login: 6.0.0 (installed 4.3.0)
  - unsplash: 2.2.1 (installed 2.2.1)
  - user_ldap: 1.19.0 (installed 1.15.0)
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
{"reqId":"AchxjP2DPwp1JltHzIx7","level":3,"time":"2024-01-16T20:30:41+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"","url":"--","message":"An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes","userAgent":"--","version":"28.0.1.1","exception":{"Exception":"Doctrine\\DBAL\\Exception\\DriverException","Message":"An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes","Code":1071,"Trace":[{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Connection.php","line":1938,"function":"convert","class":"Doctrine\\DBAL\\Driver\\API\\MySQL\\ExceptionConverter","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Connection.php","line":1880,"function":"handleDriverException","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Statement.php","line":194,"function":"convertExceptionDuringQuery","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/DB/PreparedStatement.php","line":86,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair/Collation.php","line":96,"function":"execute","class":"OC\\DB\\PreparedStatement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair.php","line":127,"function":"run","class":"OC\\Repair\\Collation","type":"->"},{"file":"/usr/local/www/nextcloud/core/Command/Maintenance/Repair.php","line":120,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Command/Command.php","line":298,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":1040,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":301,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":171,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Console/Application.php","line":206,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/occ","line":11,"args":["/usr/local/www/nextcloud/console.php"],"function":"require_once"}],"File":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php","Line":117,"Previous":{"Exception":"Doctrine\\DBAL\\Driver\\PDO\\Exception","Message":"SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes","Code":1071,"Trace":[{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php","line":132,"function":"new","class":"Doctrine\\DBAL\\Driver\\PDO\\Exception","type":"::"},{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Statement.php","line":190,"function":"execute","class":"Doctrine\\DBAL\\Driver\\PDO\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/DB/PreparedStatement.php","line":86,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair/Collation.php","line":96,"function":"execute","class":"OC\\DB\\PreparedStatement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair.php","line":127,"function":"run","class":"OC\\Repair\\Collation","type":"->"},{"file":"/usr/local/www/nextcloud/core/Command/Maintenance/Repair.php","line":120,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Command/Command.php","line":298,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":1040,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":301,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":171,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Console/Application.php","line":206,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/occ","line":11,"args":["/usr/local/www/nextcloud/console.php"],"function":"require_once"}],"File":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/PDO/Exception.php","Line":28,"Previous":{"Exception":"PDOException","Message":"SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes","Code":"42000","Trace":[{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php","line":130,"function":"execute","class":"PDOStatement","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Statement.php","line":190,"function":"execute","class":"Doctrine\\DBAL\\Driver\\PDO\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/DB/PreparedStatement.php","line":86,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair/Collation.php","line":96,"function":"execute","class":"OC\\DB\\PreparedStatement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair.php","line":127,"function":"run","class":"OC\\Repair\\Collation","type":"->"},{"file":"/usr/local/www/nextcloud/core/Command/Maintenance/Repair.php","line":120,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Command/Command.php","line":298,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":1040,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":301,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":171,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Console/Application.php","line":206,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/occ","line":11,"args":["/usr/local/www/nextcloud/console.php"],"function":"require_once"}],"File":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php","Line":130}},"message":"An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes","exception":{},"CustomMessage":"An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes"}}
{"reqId":"AchxjP2DPwp1JltHzIx7","level":3,"time":"2024-01-16T20:30:41+00:00","remoteAddr":"","user":"--","app":"no app in context","method":"","url":"--","message":"Exception while executing repair step Repair MySQL collation","userAgent":"--","version":"28.0.1.1","exception":{"Exception":"Doctrine\\DBAL\\Exception\\DriverException","Message":"An exception occurred while executing a query: SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes","Code":1071,"Trace":[{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Connection.php","line":1938,"function":"convert","class":"Doctrine\\DBAL\\Driver\\API\\MySQL\\ExceptionConverter","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Connection.php","line":1880,"function":"handleDriverException","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Statement.php","line":194,"function":"convertExceptionDuringQuery","class":"Doctrine\\DBAL\\Connection","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/DB/PreparedStatement.php","line":86,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair/Collation.php","line":96,"function":"execute","class":"OC\\DB\\PreparedStatement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair.php","line":127,"function":"run","class":"OC\\Repair\\Collation","type":"->"},{"file":"/usr/local/www/nextcloud/core/Command/Maintenance/Repair.php","line":120,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Command/Command.php","line":298,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":1040,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":301,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":171,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Console/Application.php","line":206,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/occ","line":11,"args":["/usr/local/www/nextcloud/console.php"],"function":"require_once"}],"File":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/API/MySQL/ExceptionConverter.php","Line":117,"Previous":{"Exception":"Doctrine\\DBAL\\Driver\\PDO\\Exception","Message":"SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes","Code":1071,"Trace":[{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php","line":132,"function":"new","class":"Doctrine\\DBAL\\Driver\\PDO\\Exception","type":"::"},{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Statement.php","line":190,"function":"execute","class":"Doctrine\\DBAL\\Driver\\PDO\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/DB/PreparedStatement.php","line":86,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair/Collation.php","line":96,"function":"execute","class":"OC\\DB\\PreparedStatement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair.php","line":127,"function":"run","class":"OC\\Repair\\Collation","type":"->"},{"file":"/usr/local/www/nextcloud/core/Command/Maintenance/Repair.php","line":120,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Command/Command.php","line":298,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":1040,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":301,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":171,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Console/Application.php","line":206,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/occ","line":11,"args":["/usr/local/www/nextcloud/console.php"],"function":"require_once"}],"File":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/PDO/Exception.php","Line":28,"Previous":{"Exception":"PDOException","Message":"SQLSTATE[42000]: Syntax error or access violation: 1071 Specified key was too long; max key length is 3072 bytes","Code":"42000","Trace":[{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php","line":130,"function":"execute","class":"PDOStatement","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Statement.php","line":190,"function":"execute","class":"Doctrine\\DBAL\\Driver\\PDO\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/DB/PreparedStatement.php","line":86,"function":"execute","class":"Doctrine\\DBAL\\Statement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair/Collation.php","line":96,"function":"execute","class":"OC\\DB\\PreparedStatement","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Repair.php","line":127,"function":"run","class":"OC\\Repair\\Collation","type":"->"},{"file":"/usr/local/www/nextcloud/core/Command/Maintenance/Repair.php","line":120,"function":"run","class":"OC\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Command/Command.php","line":298,"function":"execute","class":"OC\\Core\\Command\\Maintenance\\Repair","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":1040,"function":"run","class":"Symfony\\Component\\Console\\Command\\Command","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":301,"function":"doRunCommand","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/3rdparty/symfony/console/Application.php","line":171,"function":"doRun","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/lib/private/Console/Application.php","line":206,"function":"run","class":"Symfony\\Component\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/console.php","line":100,"function":"run","class":"OC\\Console\\Application","type":"->"},{"file":"/usr/local/www/nextcloud/occ","line":11,"args":["/usr/local/www/nextcloud/console.php"],"function":"require_once"}],"File":"/usr/local/www/nextcloud/3rdparty/doctrine/dbal/src/Driver/PDO/Statement.php","Line":130}},"message":"Exception while executing repair step Repair MySQL collation","exception":{},"CustomMessage":"Exception while executing repair step Repair MySQL collation"}}
```

### Additional info

FreeBSD 13.2 on amd64
Nginx 1.24.0 with php-fpm
MySQL 8.0.35
PHP 8.2.14

**Comment by joshtrichards** [comment]
Duplicate of nextcloud/mail#9247

**Comment by JRGonz** [comment]
> Duplicate of [nextcloud/mail#9247](https://github.com/nextcloud/mail/issues/9247)

Is this really the same thing? The other bug was related to an older version of nextcloud not being able to add mail indices. When I disabled mail, upgraded to 28.0.1.1, re-enabled mail, and added indices it seemed to have worked. maintenance:repair actually worked in 27 for me but I did not run into the key length error after the update to 28 when adding indices. Although... I didn't run repair on it because I didn't see a reason to do it.
